source/initial-dns-seedlist-discovery/initial-dns-seedlist-discovery.md

```diff
@@ -81,31 +105,32 @@ parse error and MUST NOT do DNS resolution or contact hosts.
 It is an error to specify more than one host name in a connection string with the `mongodb+srv` protocol, and the driver
 MUST raise a parse error and MUST NOT do DNS resolution or contact hosts.
 
-A driver MUST verify that in addition to the `{hostname}`, the `{domainname}` consists of at least two parts: the domain
-name, and a TLD. Drivers MUST raise an error and MUST NOT contact the DNS server to obtain SRV (or TXT records) if the
-full URI does not consist of at least three parts.
-
 If `mongodb+srv` is used, a driver MUST implicitly also enable TLS. Clients can turn this off by passing `tls=false` in
 either the Connection String, or options passed in as parameters in code to the MongoClient constructor (or equivalent
 API for each driver), but not through a TXT record (discussed in a later section).
 
 #### Querying DNS
 
-In this preprocessing step, the driver will query the DNS server for SRV records on `{hostname}.{domainname}`, prefixed
-with the SRV service name and protocol. The SRV service name is provided in the `srvServiceName` URI option and defaults
-to `mongodb`. The protocol is always `tcp`. After prefixing, the URI should look like:
-`_{srvServiceName}._tcp.{hostname}.{domainname}`. This DNS query is expected to respond with one or more SRV records.
+In this preprocessing step, the driver will query the DNS server for SRV records on the hostname, prefixed with the SRV
+service name and protocol. The SRV service name is provided in the `srvServiceName` URI option and defaults to
+`mongodb`. The protocol is always `tcp`. After prefixing, the URI should look like: `_{srvServiceName}._tcp.{hostname}`.
+This DNS query is expected to respond with one or more SRV records.
 
 The priority and weight fields in returned SRV records MUST be ignored.
 
 If the DNS result returns no SRV records, or no records at all, or a DNS error happens, an error MUST be raised
 indicating that the URI could not be used to find hostnames. The error SHALL include the reason why they could not be
 found.
 
-A driver MUST verify that the host names returned through SRV records have the same parent `{domainname}`. Drivers MUST
-raise an error and MUST NOT initiate a connection to any returned host name which does not share the same
+A driver MUST verify that the host names returned through SRV records share the original SRV's `{domainname}`. Drivers
+MUST raise an error and MUST NOT initiate a connection to any returned host name which does not share the same
 `{domainname}`.
 
+In the case that the SRV record name has less than three `.` separated parts, the returned address' `{domainname}` MUST
+be the SRV's entire `{hostname}` and MUST have at least one more domain level than the `{hostname}`. The next major
+version MUST no longer allow an SRV record, with any number of parts, to return an address that doesn't contain the
```

Comment:
@nbbeeken and @aditi-khare-mongoDB, regarding "The next major version MUST no longer allow an SRV record [...] to return an address that doesn't contain the SRVs' entire `{hostname}`." I just checked some SRV URIs I could see in JIRA, and I think I have some bad news on this.

It seems that Atlas hands out URIs of the form <human readable name>.<unique ID>.mongodb.net, which resolve to hostnames of the form <human readable name>-shard00-node00.<unique ID>.mongodb.net.

This suggests we're actively handing out URIs which rely on SRVs resolving to siblings. The behavior is *safe*, because different clusters' nodes are not siblings.

However, deprecation of our current behavior is probably not possible right now. I would suggest striking this text.

Reply:
Removed the extra check!